In [2]:
import pandas as pd
import datetime
import os

In [3]:
# Definir una función que aplica estilos a columnas específicas con colores diferentes
def resaltar_columnas(s, colores):
    return ['background-color: {}'.format(colores.get(col, '')) for col in s.index]

In [4]:
df = pd.read_excel("List_Obs_in_Data.xlsx") #en tu caso es un excel
df

Principal Investigator (PI)          NAME  \
0              Dr. Maximiliano Moyano (mmoyano@ucn.cl)     155989956   
1              Dr. Maximiliano Moyano (mmoyano@ucn.cl)      28051929   
2              Dr. Maximiliano Moyano (mmoyano@ucn.cl)      64034062   
3              Dr. Maximiliano Moyano (mmoyano@ucn.cl)       4646803   
4              Dr. Maximiliano Moyano (mmoyano@ucn.cl)      HIP13044   
..                                                 ...           ...   
175  Abner Zapata (Autorizado por José G. Fernández...       HD10700   
176  Abner Zapata (Autorizado por José G. Fernández...  TIC220414682   
177            Dr. Christian Moni Bidin (cmoni@ucn.cl)           ...   
178           Danilo Gonzalez (danilo.gonzalez@ucn.cl)           ...   
179                                                NaN           NaN   

         RA (J2000)    DEC (J2000)    Mag Band Exp. Time  \
0      00:29:37.415   -53:32:09.84  10.16    V      1800   
1      00:50:39.989   -18:30:21.30   8.97    V      1200   
2      02:31:59.102   -19:09:00.04  10.47    V      1800   
3    02:33:14.908\n   -10:22:42.90     10    V      1800   
4       02:47:37.44    -36:06:27.0   9.94    V      1800   
..              ...            ...    ...  ...       ...   
175    01:44:04.083  -15:56:14.927   3.50    V        ??   
176    04:44:26.849  -59:19:18.395   9.49    V      1800   
177             ...            ...    ...  ...       ...   
178             ...            ...    ...  ...       ...   
179             NaN            NaN    NaN  NaN       NaN   

                                  Comments  2021-12-21 00:00:00  \
0                               Exoplanets                    0   
1                               Exoplanets                    0   
2                               Exoplanets                    0   
3                               Exoplanets                    0   
4                               Exoplanets                    0   
..                                     ...                  ...   
175                High Proper Motion Star                    0   
176                High Proper Motion Star                    0   
177  Targets of OCs: Bacher thesis project                    0   
178                         Danilo Project                    0   
179                    # Spectra per night                   12   

     2021-12-23 00:00:00  ...  2023-07-28 00:00:00  2023-08-10 00:00:00  \
0                      0  ...                    0                    0   
1                      0  ...                    0                    0   
2                      0  ...                    0                    0   
3                      0  ...                    0                    0   
4                      0  ...                    0                    0   
..                   ...  ...                  ...                  ...   
175                    0  ...                    0                    0   
176                    0  ...                    0                    0   
177                    0  ...                   36                   10   
178                    0  ...                    0                    0   
179                    7  ...                   36                   10   

     2023-08-25 00:00:00  2023-08-29 00:00:00  2023-XX-XX  2023-XX-XX.1  \
0                      0                    0           0             0   
1                      0                    0           0             0   
2                      0                    0           0             0   
3                      0                    0           0             0   
4                      0                    0           0             0   
..                   ...                  ...         ...           ...   
175                    3                    0           0             0   
176                    0                    0           0             0   
177                    0                    0           0   

In [5]:
df.loc[df['NAME'] == "HD158902"]

Principal Investigator (PI)      NAME  \
155  Dr. José G. Fernández-Trincado (jose.fernandez...  HD158902   

       RA (J2000)    DEC (J2000)   Mag Band Exp. Time Comments  \
155  17:33:03.441  -29:39:03.037  7.21    V      1400       BH   

     2021-12-21 00:00:00  2021-12-23 00:00:00  ...  2023-07-28 00:00:00  \
155                    0                    0  ...                    0   

     2023-08-10 00:00:00  2023-08-25 00:00:00  2023-08-29 00:00:00  \
155                    0                    1                    0   

     2023-XX-XX  2023-XX-XX.1  2023-XX-XX.2  2023-XX-XX.3 2023-XX-XX.4  \
155           0             0             0             0            0   

     Unnamed: 109  
155             1  

[1 rows x 110 columns]

#### Filtrado datos de Dr. Christian Moni Bidin (cmoni@ucn.cl)

Filtramos aquellos que tienen 3 o más columnas consecutivas.

Las columnas resaltadas en colores representan días consecutivos.

In [6]:

# Leer el archivo Excel
df = pd.read_excel("List_Obs_in_Data.xlsx")

# Filtrar el DataFrame por el "Principal Investigator (PI)"
df_christian = df[df["Principal Investigator (PI)"] == "Dr. Christian Moni Bidin (cmoni@ucn.cl)"]

# Convertir todas las columnas relevantes a tipo numérico (ignorar errores)
# Especificamos columns=slice(start, end) para seleccionar solo las columnas que queremos convertir
df_christian.iloc[:, 8:] = df_christian.iloc[:, 8:].apply(pd.to_numeric, errors='coerce')

# Función para verificar si hay 5 valores consecutivos diferentes de cero en una fila
def fechas_consecutivas(row):
    count = 0
    for value in row:
        if value != 0 and not pd.isna(value):
            count += 1
            if count >= 3:
                return True
        else:
            count = 0
    return False

# Aplicar la función a cada fila del DataFrame
df_christian_filtered = df_christian[df_christian.iloc[:, 8:].apply(fechas_consecutivas, axis=1)]


# Seleccionar columnas que tienen al menos un valor diferente de cero
pd.set_option('display.max_columns', None)
non_zero_columns_christian = df_christian_filtered.loc[:, (df_christian_filtered != 0).any(axis=0)]

# Especificar los colores para cada columna
colores = ["#D4C685","#F7EF81","#CFE795","#A7D3A6","#ADD2C2"]
colores_resaltar = {datetime.datetime(2022, 4, 21, 0, 0): colores[0],datetime.datetime(2022, 4, 22, 0, 0): colores[0],datetime.datetime(2022, 4, 23, 0, 0): colores[0],datetime.datetime(2023, 3, 30, 0, 0): colores[1],datetime.datetime(2023, 3, 31, 0, 0): colores[1],datetime.datetime(2023, 7, 27, 0, 0): colores[2],datetime.datetime(2023, 7, 28, 0, 0): colores[2]}

# Aplicar estilos al DataFrame
estilos = non_zero_columns_christian.style.apply(lambda x: resaltar_columnas(x, colores_resaltar), axis=1)

estilos

| Name | Observación |
| :--- | :--- |
| 46388 | 3 noches consecutivas |
| 50288 | 3 noches consecutivas |
| 54718 | 3 noches consecutivas |
| 59281 | 3 noches consecutivas |
| 61959 | 3 noches consecutivas |
| 68380 | 3 noches consecutivas |
| 70693 | 3 noches consecutivas |
| 72217 | 3 noches consecutivas |
| Fila 177 | 2 pares de noches consecutivas con muchos datos por noche |

#### Filtrado datos de Dr. José G. Fernández-Trincado (jose.fernandez@ucn.cl)

Filtramos aquellos que tienen 3 o más columnas consecutivas.

Las columnas resaltadas en colores representan días consecutivos.

In [7]:
# Filtrar el DataFrame por el "Principal Investigator (PI)"
df_jose = df[df["Principal Investigator (PI)"] == "Dr. José G. Fernández-Trincado (jose.fernandez@ucn.cl)"]

# Convertir todas las columnas relevantes a tipo numérico (ignorar errores)
# Especificamos columns=slice(start, end) para seleccionar solo las columnas que queremos convertir
df_jose.iloc[:, 8:] = df_jose.iloc[:, 8:].apply(pd.to_numeric, errors='coerce')

# Función para verificar si hay 5 valores consecutivos diferentes de cero en una fila
def fechas_consecutivas(row):
    count = 0
    for value in row:
        if value != 0 and not pd.isna(value):
            count += 1
            if count >= 3:
                return True
        else:
            count = 0
    return False

# Filtrando las observaciones de nuestros objetos de estudio
df_jose = df_jose[df_jose['NAME'] != 'Nrich_2']
df_jose = df_jose[df_jose['NAME'] != 'Nrich_3']
df_jose = df_jose[df_jose['NAME'] != 'Nrich_5']
df_jose = df_jose[df_jose['NAME'] != 'Nrich_7']
df_jose = df_jose[df_jose['NAME'] != 's-rich']
df_jose = df_jose[df_jose['NAME'] != 'Si_rich']
df_jose = df_jose[df_jose['NAME'] != 'Krich_2']


# Aplicar la función a cada fila del DataFrame
df_jose_filtered = df_jose[df_jose.iloc[:, 8:].apply(fechas_consecutivas, axis=1)]


# Seleccionar columnas que tienen al menos un valor diferente de cero
pd.set_option('display.max_columns', None)
non_zero_columns_jose = df_jose_filtered.loc[:, (df_jose_filtered != 0).any(axis=0)]
non_zero_columns_jose
# # Especificar los colores para cada columna
colores = ["#D4C685","#F7EF81","#CFE795","#A7D3A6","#ADD2C2"]
colores_resaltar = {datetime.datetime(2022, 3, 3, 0, 0): colores[0],
                    datetime.datetime(2022, 3, 4, 0, 0): colores[0],
                    datetime.datetime(2022, 3, 5, 0, 0): colores[0],
                    datetime.datetime(2022, 3, 17, 0, 0): colores[1],
                    datetime.datetime(2022, 3, 18, 0, 0): colores[1],
                    datetime.datetime(2022, 3, 19, 0, 0): colores[1],
                    datetime.datetime(2022, 3, 20, 0, 0): colores[1],
                    datetime.datetime(2022, 3, 25, 0, 0): colores[2],
                    datetime.datetime(2022, 3, 26, 0, 0): colores[2],
                    datetime.datetime(2022, 3, 27, 0, 0): colores[2]}

# # Aplicar estilos al DataFrame
estilos = non_zero_columns_jose.style.apply(lambda x: resaltar_columnas(x, colores_resaltar), axis=1)

estilos

| Etiqueta | Significado |
|:---|:---|
|⭐| Que tiene 3 y 4 noches consecutivas y además con al menos una noche de más de 1 dato|
|🚩🚩| Que tiene 3 noches consecutivas y además por lo menos una noche con más de 1 dato|
|🚩| Que tiene 3 noches consecutivas (de un dato c/u) y además 2 noches consecutivas con al menos una noche con más de 1 dato|


| Name | Observación | Etiqueta|
| :--- | :--- | :--- |
| FR_CMa | 3 noches consecutivas |  |
| kap_CMa | 3 noches consecutivas | 🚩 |
| omi_Pup | 3 y 4 noches consecutivas | ⭐ |
| V916_Cen | 3 noches consecutivas | 🚩 |
| BZ_Cru | 3 y 4 noches consecutivas | ⭐ |
| HD_112147 | 3 noches consecutivas | 🚩🚩  |
| LU_Mus | 3 noches consecutivas | 🚩  |
| GSE_2 | 3 noches consecutivas | 🚩🚩 |
| GSE_5 | 3 noches consecutivas | 🚩🚩  |

#### Filtrado datos de Abner Zapata (Autorizado por José G. Fernández-Trincado)

Las columnas resaltadas en colores representan días consecutivos.

In [8]:
# Filtrar el DataFrame por el "Principal Investigator (PI)"
df_abner = df[df["Principal Investigator (PI)"] == "Abner Zapata (Autorizado por José G. Fernández-Trincado)"]

# Convertir todas las columnas relevantes a tipo numérico (ignorar errores)
# Especificamos columns=slice(start, end) para seleccionar solo las columnas que queremos convertir
df_abner.iloc[:, 8:] = df_abner.iloc[:, 8:].apply(pd.to_numeric, errors='coerce')

# Seleccionar columnas que tienen al menos un valor diferente de cero
pd.set_option('display.max_columns', None)
non_zero_columns_abner = df_abner.loc[:, (df_abner != 0).any(axis=0)]
non_zero_columns_abner
# # Especificar los colores para cada columna
colores = ["#D4C685","#F7EF81","#CFE795","#A7D3A6","#ADD2C2"]
colores_resaltar = {datetime.datetime(2022, 11, 25, 0, 0): colores[3],
                    datetime.datetime(2022, 11, 26, 0, 0): colores[3]}

# # # Aplicar estilos al DataFrame
estilos = non_zero_columns_abner.style.apply(lambda x: resaltar_columnas(x, colores_resaltar), axis=1)

estilos

**<span style="color:red"> Nota:</span>** *<span style="color:red">La estrella HD10700 fué una de las estrellas estándar que nos mencionó Joanne</span>*

In [9]:
# Esta es otra estrella estándar que mencionó Joanne, pero no está en este conjunto de datos con ese nombre.
una = df[df["NAME"] == "HIP104476"]
una

Empty DataFrame
Columns: [Principal Investigator (PI), NAME, RA (J2000), DEC (J2000), Mag, Band, Exp. Time, Comments, 2021-12-21 00:00:00, 2021-12-23 00:00:00, 2022-01-04 00:00:00, 2022-01-06 00:00:00, 2022-01-07 00:00:00, 2022-01-11 00:00:00, 2022-01-13 00:00:00, 2022-01-15 00:00:00, 2022-01-16 00:00:00, 2022-01-25 00:00:00, 2022-01-27 00:00:00, 2022-01-29 00:00:00, 2022-02-04 00:00:00, 2022-02-05 00:00:00, 2022-03-01 00:00:00, 2022-03-03 00:00:00, 2022-03-04 00:00:00, 2022-03-05 00:00:00, 2022-03-11 00:00:00, 2022-03-17 00:00:00, 2022-03-18 00:00:00, 2022-03-19 00:00:00, 2022-03-20 00:00:00, 2022-03-22 00:00:00, 2022-03-25 00:00:00, 2022-03-26 00:00:00, 2022-03-27 00:00:00, 2022-04-12 00:00:00, 2022-04-19 00:00:00, 2022-04-21 00:00:00, 2022-04-22 00:00:00, 2022-04-23 00:00:00, 2022-04-28 00:00:00, 2022-04-29 00:00:00, 2022-04-30 00:00:00, 2022-05-03 00:00:00, 2022-05-10 00:00:00, 2022-05-12 00:00:00, 2022-05-13 00:00:00, 2022-05-14 00:00:00, 2022-06-16 (Bad Weather), 2022-09-01 00:00:00, 2022-09-02 00:00:00, 2022-09-03 00:00:00, 2022-09-06 00:00:00, 2022-09-13 00:00:00, 2022-09-15 00:00:00, 2022-09-29 00:00:00, 2022-09-30 00:00:00, 2022-10-01 00:00:00, 2022-10-04 00:00:00, 2022-10-06 00:00:00, 2022-10-11 00:00:00, 2022-10-13 00:00:00, 2022-10-18 00:00:00, 2022-10-20 00:00:00, 2022-10-21 00:00:00, 2022-11-03 00:00:00, 2022-11-04 00:00:00, 2022-11-22 00:00:00, 2022-11-25 00:00:00, 2022-11-26 00:00:00, 2022-12-01 00:00:00, 2022-12-02 00:00:00, 2022-12-06 00:00:00, 2022-12-09 00:00:00, 2022-12-13 00:00:00, 2022-12-16 00:00:00, 2022-12-20 00:00:00, 2022-12-21 00:00:00, 2023-01-10 00:00:00, 2023-03-16 00:00:00, 2023-03-23 00:00:00, 2023-03-30 00:00:00, 2023-03-31 00:00:00, 2023-04-11 00:00:00, 2023-04-21 00:00:00, 2023-04-22 00:00:00, 2023-04-24 00:00:00, 2023-04-27 00:00:00, 2023-04-28 00:00:00, 2023-05-02 00:00:00, 2023-05-04 00:00:00, 2023-05-23 00:00:00, 2023-05-26 00:00:00, 2023-05-27 00:00:00, 2023-06-02 00:00:00, 2023-06-08 00:00:00, 2023-06-15 00:00:00, 2023-07-13 00:00:00, 2023-07-14 00:00:00, 2023-07-27 00:00:00, ...]
Index: []

#### Filtrado datos de Danilo Gonzalez (danilo.gonzalez@ucn.cl)

Las columnas resaltadas en colores representan días consecutivos.

In [10]:
# Filtrar el DataFrame por el "Principal Investigator (PI)"
df_abner = df[df["Principal Investigator (PI)"] == "Danilo Gonzalez (danilo.gonzalez@ucn.cl)"]

# Convertir todas las columnas relevantes a tipo numérico (ignorar errores)
# Especificamos columns=slice(start, end) para seleccionar solo las columnas que queremos convertir
df_abner.iloc[:, 8:] = df_abner.iloc[:, 8:].apply(pd.to_numeric, errors='coerce')

# Seleccionar columnas que tienen al menos un valor diferente de cero
pd.set_option('display.max_columns', None)
non_zero_columns_abner = df_abner.loc[:, (df_abner != 0).any(axis=0)]
non_zero_columns_abner
# # # Especificar los colores para cada columna
# colores = ["#D4C685","#F7EF81","#CFE795","#A7D3A6","#ADD2C2"]
# colores_resaltar = {datetime.datetime(2022, 11, 25, 0, 0): colores[3],
#                     datetime.datetime(2022, 11, 26, 0, 0): colores[3]}

# # # # Aplicar estilos al DataFrame
# estilos = non_zero_columns_abner.style.apply(lambda x: resaltar_columnas(x, colores_resaltar), axis=1)

# estilos

Principal Investigator (PI) NAME RA (J2000) DEC (J2000)  \
178  Danilo Gonzalez (danilo.gonzalez@ucn.cl)  ...        ...         ...   

     Mag Band Exp. Time        Comments  2023-04-11 00:00:00  \
178  ...  ...       ...  Danilo Project                    8   

     2023-05-26 00:00:00  2023-05-27 00:00:00  2023-06-02 00:00:00  \
178                    2                    4                    4   

     2023-06-08 00:00:00  Unnamed: 109  
178                    3            21

In [11]:
# Obtén la ruta absoluta
file_path = os.path.abspath('Nrich/news_RV/Objetos_UCN.csv')
print(file_path)

# Luego intenta leer el archivo
df_UCN = pd.read_csv(file_path)
df_UCN

/home/alexandert/Docs_Jupyter/9.Joven_Investigador/Nrich/news_RV/Objetos_UCN.csv


Fecha    Objetos  Rep
0    2022-01-05    HD38120    1
1    2022-03-02   HD112147    1
2    2022-03-02    HD93683    1
3    2022-03-05   HD112147    1
4    2022-03-05    HD93683    1
..          ...        ...  ...
261  2023-08-11    HIP9774    1
262  2023-08-11  HIP101955    1
263  2023-08-11  HIP104476    1
264  2023-08-25   HIP70693    1
265  2023-08-26    HD10700    3

[266 rows x 3 columns]

In [12]:
# Convertir la columna de fechas a tipo datetime
df_UCN['Fecha'] = pd.to_datetime(df_UCN['Fecha'])

# Agrupar por objeto y obtener las fechas en las que se encuentran
grouped = df_UCN.groupby("Objetos")['Fecha'].apply(list).reset_index()

# Mostrar el resultado
grouped

Objetos                                              Fecha
0        HD101545                              [2022-03-06 00:00:00]
1        HD101964                              [2023-04-12 00:00:00]
2        HD101993                              [2023-04-12 00:00:00]
3        HD101994                              [2023-04-12 00:00:00]
4        HD102009                              [2023-04-12 00:00:00]
..            ...                                                ...
104  TIC219345200         [2022-12-03 00:00:00, 2022-12-07 00:00:00]
105   TIC28051929                              [2022-09-14 00:00:00]
106  TIC328321923                              [2022-12-03 00:00:00]
107    TIC4646803                              [2022-12-03 00:00:00]
108   TIC64034062  [2022-09-14 00:00:00, 2022-12-03 00:00:00, 202...

[109 rows x 2 columns]

In [13]:
conteo_nombres = df_UCN["Objetos"].value_counts()
conteo_nombres

Objetos
HD72673      15
HD93683      13
HD112147     11
HD157347      7
HD129116      6
             ..
HIP57421      1
HIP61158      1
HIP58669      1
HIP56319      1
HIP104476     1
Name: count, Length: 109, dtype: int64

In [14]:
# Agrupar por la columna 'Objetos' y sumar los valores de la columna 'Rep'
suma_reps_por_objeto = df_UCN.groupby("Objetos")["Rep"].sum()

# Mostrar el resultado
df_conteo = suma_reps_por_objeto.reset_index()

# Renombrar las columnas para mayor claridad si es necesario
df_conteo.columns = ["Objetos", 'Suma_Rep']
df_conteo = df_conteo.sort_values(by='Suma_Rep', ascending=False)
df_conteo = df_conteo[df_conteo["Suma_Rep"] != 1]
df_conteo

Objetos  Suma_Rep
16        HD72673        28
6        HD112147        22
8        HD157347        16
17        HD93683        13
7        HD129116        12
96       HIP91880         6
5         HD10700         6
100      HIP97690         6
71       HIP68380         5
98       HIP96754         5
65       HIP59281         5
95       HIP90198         5
58       HIP54718         5
19      HIP100998         5
99       HIP97365         4
21      HIP101955         4
49       HIP42916         4
75       HIP72492         4
51       HIP46388         4
104  TIC219345200         4
55       HIP50288         4
12       HD308889         4
72       HIP69160         4
73       HIP70693         4
50       HIP45726         3
32      HIP111898         3
93       HIP86032         3
94       HIP86142         3
40       HIP16083         3
83       HIP78401         3
36       HIP12225         3
33      HIP111974         3
108   TIC64034062         3
31      HIP110960         3
25      HIP103987         3
11       HD308888         3
103  TIC168656470         3
102  TIC155989956         2
74       HIP72217         2
69       HIP62985         2
79       HIP76031         2
9        HD222925         2
63       HIP57894         2
10       HD308887         2
60       HIP57283         2
57       HIP53206         2
13       HD309001         2
34        HIP1144         2
18       HIP10096         2
87       HIP81874         2
27      HIP105382         2
101       HIP9774         2
29      HIP109908         2
28      HIP106978         2
47       HIP40523         2
23       HIP10305         2
42       HIP23776         2
38       HIP13769         2
92       HIP84970         2
46       HIP38146         2

In [15]:
df_conteo.to_csv('probables_estándar.csv', index=False)

In [16]:
pd.set_option('display.max_colwidth', None)
df_conteo_obs = pd.read_csv('probables_estándar_info.csv')
df_conteo_obs

Objetos  Suma_Rep                      Tipo  \
0        HD72673        28   High Proper Motion Star   
1       HD112147        22                   Be Star   
2       HD157347        16   High Proper Motion Star   
3        HD93683        13                   Be Star   
4       HD129116        12   Double or Multiple Star   
5       HIP91880         6   Double or Multiple Star   
6        HD10700         6   High Proper Motion Star   
7       HIP97690         6   High Proper Motion Star   
8       HIP68380         5   Double or Multiple Star   
9       HIP96754         5   High Proper Motion Star   
10      HIP59281         5   Double or Multiple Star   
11      HIP90198         5                      Star   
12      HIP54718         5   Double or Multiple Star   
13     HIP100998         5                      Star   
14      HIP97365         4      Spectroscopic Binary   
15     HIP101955         4           BY Dra Variable   
16      HIP42916         4   Double or Multiple Star   
17      HIP72492         4   Double or Multiple Star   
18      HIP46388         4      Spectroscopic Binary   
19  TIC219345200         4      Spectroscopic Binary   
20      HIP50288         4   Double or Multiple Star   
21      HD308889         4        Emission-line Star   
22      HIP69160         4   Double or Multiple Star   
23      HIP70693         4   High Proper Motion Star   
24      HIP45726         3   Double or Multiple Star   
25     HIP111898         3   Double or Multiple Star   
26      HIP86032         3   Double or Multiple Star   
27      HIP86142         3   Double or Multiple Star   
28      HIP16083         3   High Proper Motion Star   
29      HIP78401         3      Spectroscopic Binary   
30      HIP12225         3   Double or Multiple Star   
31     HIP111974         3      Spectroscopic Binary   
32   TIC64034062         3      Spectroscopic Binary   
33     HIP110960         3   Double or Multiple Star   
34     HIP103987         3      Spectroscopic Binary   
35      HD308888         3                      Star   
36  TIC168656470         3                      Star   
37  TIC155989956         2                      Star   
38      HIP72217         2   Double or Multiple Star   
39      HIP62985         2      Long-Period Variable   
40      HIP76031         2      Spectroscopic Binary   
41      HD222925         2  Chemically Peculiar Star   
42      HIP57894         2   Double or Multiple Star   
43      HD308887         2                      Star   
44      HIP57283         2   Double or Multiple Star   
45      HIP53206         2   Double or Multiple Star   
46      HD309001         2                      Star   
47       HIP1144         2   Double or Multiple Star   
48      HIP10096         2      Spectroscopic Binary   
49      HIP81874         2                      Star   
50     HIP105382         2       alpha2 CVn Variable   
51       HIP9774         2   Double or Multiple Star   
52     HIP109908         2   Double or Multiple Star   
53     HIP106978         2   Double or Multiple Star   
54      HIP40523         2   High Proper Motion Star   
55      HIP10305         2      Spectroscopic Binary   
56      HIP23776         2   High Proper Motion Star   
57      HIP13769         2      Spectroscopic Binary   
58      HIP84970         2         beta Cep Variable   
59      HIP38146         2   Double or Multiple Star   

                                                                             Observación  \
0                                                    En simbad aparecen 16 valores de RV   
1            Posible estándar de velocidad radial, en simbad aparecen 4 valores de -29.0   
2                                                    En simbad aparecen 14 valores de RV   
3                                                    No presenta valores de RV en simbad   
4                                                            Diferencias de 8 km/s en RV   
5                                       

In [17]:
filtro1 = df_conteo_obs["Tipo"] != "Double or Multiple Star"
df_conteo_filt = df_conteo_obs[filtro1]
filtro2 = df_conteo_filt["Tipo"] != "Spectroscopic Binary"
df_conteo_filt = df_conteo_filt[filtro2]
df_conteo_filt

Objetos  Suma_Rep                      Tipo  \
0        HD72673        28   High Proper Motion Star   
1       HD112147        22                   Be Star   
2       HD157347        16   High Proper Motion Star   
3        HD93683        13                   Be Star   
6        HD10700         6   High Proper Motion Star   
7       HIP97690         6   High Proper Motion Star   
9       HIP96754         5   High Proper Motion Star   
11      HIP90198         5                      Star   
13     HIP100998         5                      Star   
15     HIP101955         4           BY Dra Variable   
21      HD308889         4        Emission-line Star   
23      HIP70693         4   High Proper Motion Star   
28      HIP16083         3   High Proper Motion Star   
35      HD308888         3                      Star   
36  TIC168656470         3                      Star   
37  TIC155989956         2                      Star   
39      HIP62985         2      Long-Period Variable   
41      HD222925         2  Chemically Peculiar Star   
43      HD308887         2                      Star   
46      HD309001         2                      Star   
49      HIP81874         2                      Star   
50     HIP105382         2       alpha2 CVn Variable   
54      HIP40523         2   High Proper Motion Star   
56      HIP23776         2   High Proper Motion Star   
58      HIP84970         2         beta Cep Variable   

                                                                             Observación  \
0                                                    En simbad aparecen 16 valores de RV   
1            Posible estándar de velocidad radial, en simbad aparecen 4 valores de -29.0   
2                                                    En simbad aparecen 14 valores de RV   
3                                                    No presenta valores de RV en simbad   
6         En simbad aparecen 26 valores de RV (Es la estrella que tenemos como estándar)   
7              Posible estándar de velocidad radial, en simbad aparecen 2 valores de 6.4   
9   Posible estándar de velocidad radial, en simbad aparecen 3 valores de -17,36 y -17,6   
11                                                   No presenta valores de RV en simbad   
13                          En simbad aparecen 3 valores de RV (Diferencias de 0,7 km/s)   
15                                                          Diferencias de 29 km/s en RV   
21                                                   No presenta valores de RV en simbad   
23                         En simbad aparecen 3 valores de RV (Diferencias de 0,17 km/s)   
28                                                    En simbad aparecen 5 valores de RV   
35                                                   No presenta valores de RV en simbad   
36                                                                                   NaN   
37                           En simbad aparecen 3 valores de RV (Diferencias de 13 km/s)   
39                                                        Diferencias de 5.18 km/s en RV   
41                                                    En simbad aparecen 5 valores de RV   
43                                                   No presenta valores de RV en simbad   
46                                       Solo tiene un valor de RV en simbad de -18 km/s   
49                                      En simbad aparecen 2 valores de RV de -88.5 km/s   
50                                                         Diferencias de 5.6 km/s de RV   
54                                                         Diferencias de 0.1 km/s en RV   
56                                                           Diferencias de 5 km/s en RV   
58                                                        Diferencias de 18.8 km/s de RV   

                                  Parents  
0                                     NaN  
1                            Open Cluster  
2                          

In [18]:
grouped[grouped["Objetos"] == "HD72673"]

Objetos  \
16  HD72673   

                                                                                                                                                                                                                                                                                                                          Fecha  
16  [2022-04-20 00:00:00, 2022-04-22 00:00:00, 2022-04-23 00:00:00, 2022-04-24 00:00:00, 2022-04-29 00:00:00, 2022-04-30 00:00:00, 2022-05-01 00:00:00, 2022-05-04 00:00:00, 2022-05-06 00:00:00, 2022-05-11 00:00:00, 2022-05-13 00:00:00, 2022-05-14 00:00:00, 2023-03-17 00:00:00, 2023-03-24 00:00:00, 2023-04-01 00:00:00]

In [19]:
grouped[grouped["Objetos"] == "HD112147"]

Objetos  \
6  HD112147   

                                                                                                                                                                                                                                     Fecha  
6  [2022-03-02 00:00:00, 2022-03-05 00:00:00, 2022-03-12 00:00:00, 2022-03-19 00:00:00, 2022-03-20 00:00:00, 2022-03-21 00:00:00, 2022-03-23 00:00:00, 2022-03-26 00:00:00, 2022-03-27 00:00:00, 2023-04-28 00:00:00, 2023-04-29 00:00:00]

In [20]:
grouped[grouped["Objetos"] == "HD157347"]

Objetos  \
8  HD157347   

                                                                                                                                                 Fecha  
8  [2022-04-13 00:00:00, 2023-04-22 00:00:00, 2023-04-22 00:00:00, 2023-04-23 00:00:00, 2023-04-28 00:00:00, 2023-04-29 00:00:00, 2023-06-03 00:00:00]

In [21]:
grouped[grouped["Objetos"] == "HD93683"]

Objetos  \
17  HD93683   

                                                                                                                                                                                                                                                                                Fecha  
17  [2022-03-02 00:00:00, 2022-03-05 00:00:00, 2022-03-06 00:00:00, 2022-03-12 00:00:00, 2022-03-21 00:00:00, 2022-03-27 00:00:00, 2022-03-28 00:00:00, 2022-04-13 00:00:00, 2022-04-20 00:00:00, 2023-04-23 00:00:00, 2023-04-28 00:00:00, 2023-04-29 00:00:00, 2023-05-03 00:00:00]

In [22]:
grouped[grouped["Objetos"] == "HD10700"]

Objetos                                       Fecha
5  HD10700  [2023-07-15 00:00:00, 2023-08-26 00:00:00]

In [25]:
grouped[grouped["Objetos"] == "HIP100998"]

Objetos  \
19  HIP100998   

                                                                                                        Fecha  
19  [2022-09-14 00:00:00, 2023-04-25 00:00:00, 2023-05-24 00:00:00, 2023-06-16 00:00:00, 2023-07-14 00:00:00]